code/__HELPERS/game.dm

```diff
@@ -11,6 +11,33 @@
 		return
 	return format_text ? format_text(A.name) : A.name
 
+/proc/IsValidForResinStructure(turf/target, needsSupport = FALSE)
+	if(!target || !istype(target))
+		return 0
+	var/obj/alien/weeds/alien_weeds = locate() in target
+	if(!target.check_disallow_alien_fortification(null, TRUE))
+		return "Not allowed to build here"
```

Comment:
Sometimes you return a bool, sometimes a boolean.

You can probably pass the builder as an optional, and do a to chat if it is not null